ubcpi/ubcpi.py

```diff
@@ -0,0 +1,468 @@
+        Returns:
+            unicode
+
+        """
+        if hasattr(key, 'to_deprecated_string'):
+            return key.to_deprecated_string()
+        else:
+            return unicode(key)
+
+
+@XBlock.needs('user')
+class PeerInstructionXBlock(XBlock, MissingDataFetcherMixin):
+    """
+    Notes:
+    storing index vs option text: when storing index, it is immune to the
+    option text changes. But when the order changes, the results will be
+    skewed. When storing option text, it is impossible (at least for now) to
+    update the existing students responses when option text changed.
+
+    a warning may be shown to the instructor that they may only do the minor
+    changes to the question options and may not change the order of the options,
+    add or delete options
+    """
+
+    # Fields are defined on the class.  You can access them in your code as
+    # self.<fieldname>.
+
+    # the display name that used on the interface
+    display_name = String(default="Peer Instruction")
+
+    question_text = Dict(
+    	default={'text': 'What is the answer to life, the universe and everything?', 'image_url': '', 'image_position': 'below', 'show_image_fields': 0, 'image_alt': ''}, scope=Scope.content,
+    	help="The question the students see. This question appears above the possible answers which you set below. You can use text, an image or a combination of both. If you wish to add an image to your question, press the 'Add Image' button."
+    )
+
+    options = List(
+        # default=['Default Option 1', 'Default Option 2'], scope=Scope.content,
+        default=[
+            {'text': '21', 'image_url': '', 'image_position': 'below', 'show_image_fields': 0, 'image_alt': ''},
+            {'text': '42', 'image_url': '', 'image_position': 'below', 'show_image_fields': 0, 'image_alt': ''},
+            {'text': '63', 'image_url': '', 'image_position': 'below', 'show_image_fields': 0, 'image_alt': ''}
+        ],
+        scope=Scope.content,
+        help="The possible options from which the student may select",
+    )
+
+    rationale_size = Dict(
+        default={'min': 1, 'max': MAX_RATIONALE_SIZE}, scope=Scope.content,
+        help="The minimum and maximum number of characters a student is allowed for their rationale.",
+    )
+
+    correct_answer = Integer(
+        default=1, scope=Scope.content,
+        help="The correct option for the question",
+    )
+
+    correct_rationale = Dict(
+        default={'text': "In the radio series and the first novel, a group of hyper-intelligent pan-dimensional beings demand to learn the Answer to the Ultimate Question of Life, The Universe, and Everything from the supercomputer, Deep Thought, specially built for this purpose. It takes Deep Thought 7.5 million years to compute and check the answer, which turns out to be 42. Deep Thought points out that the answer seems meaningless because the beings who instructed it never actually knew what the Question was." }, scope=Scope.content,
```

Comment:
This seems like a very long string of text to present as the default for every block.